angular/projects/global/src/lib/services/store.service.ts

```diff
@@ -0,0 +1,153 @@
+export class StoreService {
+  private currentState: any;
+
+  constructor() {
+    this.currentState = {};
+    this.state.navigationStack = [];
+  }
+
+  get state(): any {
+    return this.currentState;
+  }
+  set state(state: any) {
+    this[state.type] = state.data;
+  }
+
+  public getNavigationStack() {
+    if (this.state.navigationStack) {
+      return this.state.navigationStack;
+    } else {
+      return null;
+    }
+  }
+
+  public getLastBackUrl() {
+    const stack = this.getNavigationStack();
+    if (stack) {
+      return stack[stack.length - 1].backUrl;
+    } else {
+      return null;
+    }
+  }
+
+  public getLastNavigationObject() {
+    const stack = this.getNavigationStack();
+    if (stack) {
+      return stack[stack.length - 1];
+    } else {
+      return null;
+    }
+  }
+
+  public pushNavigationStack(backUrl: string[], breadcrumbs: object[], componentId: string = null) {
+    const navigationObject = {
+      backUrl: backUrl,
+      breadcrumbs: breadcrumbs,
+      componentId: componentId
+    };
```

Comment:
It would be good to turn this into a type (interface), and apply it anywhere its relevant.

Then additionally, instead of expecting 3 parameters in this method, you could just take in a one parameter. 
Something like:
`public pushNavigationStack(navigaionObject: INavigationObject) {`

Reply:
Just read further down and see that there already is a navigation type.  So maybe just needs to be used here where relevant.